lib/utility/recipe_puppy_parser.rb

```diff
@@ -0,0 +1,92 @@
+module Utility
+  class RecipePuppyParser
+    ALL_RECIPES_URL = 'http://www.recipepuppy.com/api/?q=&p='.freeze
+    BASE_URL = 'http://www.recipepuppy.com/api/'.freeze
+
+# API/db scrape/validation PARENT 1
+    def query_all_recipes_for(*ingredients)
+      ingredient_query_syntax = ingredients.join('').gsub(/\s+/, "")
+      contact_recipe_api(ingredient_query_syntax)
+    end
+
+    def contact_recipe_api(ingredient_query)
+      recipes = []
+      100.times do |page|
+        searched_url = "#{BASE_URL}?i=#{ingredient_query}&p=#{page + 1}"
+        get_recipes(recipes, searched_url)
+      end
+      return recipes
+    end
+
+    def get_recipes_to_query(recipe_array, url)
+      response = HTTParty.get(url)
+      return error unless response.success?
+        parsed = JSON.parse(response)
+        parsed["results"].each do |recipe|
+          organise_new_vs_old_recipes(recipe)
+      end
+      recipe_array
+    end
+
+    def organise_new_vs_old_recipes(results_recipe)
+      recipe_ingredients = results_recipe["ingredients"].split(', ')
+      # FIX, dont want to get rid of whitespace
+      recipe_name_no_special_characters = results_recipe["title"].gsub!(/[^0-9A-Za-z]/, '')
+      check_recipes = Recipe.find_by(name: recipe_name_no_special_characters)
+        process_ingredients_for(check_recipes, recipe_ingredients) if check_recipes
+      new_recipe = Recipe.create({name: recipe_name_no_special_characters, thumbnail: results_recipe["thumbnail"]})
+        process_ingredients_for(new_recipe, recipe_ingredients) if new_recipe
+    end
+
+# Complete Scrape PARENT
+    def save_all_recipes_to_database
+      recipes = []
```

Comment:
you create this array to pass into get_recipes_to_save, but it doesn't seem to be used in that method. Could this be left out entirely?